temporal-sdk/src/main/java/io/temporal/client/LazyUpdateHandleImpl.java

```diff
@@ -73,18 +73,18 @@ public String getId() {
 
   @Override
   public CompletableFuture<T> getResultAsync(long timeout, TimeUnit unit) {
-    WorkflowClientCallsInterceptor.PollWorkflowUpdateOutput<T> pollCall;
-    if (previousPollCall == null) {
-      pollUntilComplete(timeout, unit);
-    }
-    pollCall = previousPollCall;
-    // Get result may be called more than once for non-complete wait policies, so reset it here.
-    previousPollCall = null;
+
+    WorkflowClientCallsInterceptor.PollWorkflowUpdateOutput<T> pollCall =
+        pollUntilComplete(timeout, unit);
 
     return pollCall
         .getResult()
         .exceptionally(
             failure -> {
+              // If the poll didn't find the completion successfully, reset the previous poll call
```

Comment:
Just because the poll call failed does not mean it shouldn't be cached right? A update rejection would also complete the poll future exceptional If I recall correctly. I would probably drop caching since we don't cache workflow result either and if we get user feedback address all these functions with a consistent strategy.

Reply:
Sure, I could just delete it every time still. I need something like it for the don't-return-handle-until-completed case and I figured why not cache it for the success case since that's definitely not going to change.

Reply:
I think for the don't-return-handle-until-completed case you can take the result and put it in the `CompletedUpdateHandleImpl`

Reply:
Sure, that works for me

Reply:
K, I've done a much more targeted version of this.

Reply:
We still do want to avoid caching most exceptions though, just the ones from the update outcome should be cached